Src/ILGPU/IR/Values/LanguageValues.cs

```diff
@@ -0,0 +1,216 @@
+﻿// ---------------------------------------------------------------------------------------
+//                                        ILGPU
+//                        Copyright (c) 2016-2020 Marcel Koester
+//                                    www.ilgpu.net
+//
+// File: LanguageValues.cs
+//
+// This file is part of ILGPU and is distributed under the University of Illinois Open
+// Source License. See LICENSE.txt for details
+// ---------------------------------------------------------------------------------------
+
+using ILGPU.IR.Construction;
+using ILGPU.IR.Types;
+using ILGPU.Runtime;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using FormatArray = System.Collections.Immutable.ImmutableArray<
+    ILGPU.Util.FormatString.FormatExpression>;
+using ValueList = ILGPU.Util.InlineList<ILGPU.IR.Values.ValueReference>;
+
+namespace ILGPU.IR.Values
+{
+    /// <summary>
+    /// Represents the kind of inline language.
+    /// </summary>
+    public enum LanguageKind
+    {
+        /// <summary>
+        /// Inline PTX assembly.
+        /// </summary>
+        PTX,
+    }
+
+    /// <summary>
+    /// Represents the <see cref="Accelerator.LanguageVersion"/> property.
+    /// </summary>
+    [ValueKind(ValueKind.LanguageVersion)]
+    public sealed class LanguageVersionValue : DeviceConstantValue
+    {
+        #region Instance
+
+        /// <summary>
+        /// Constructs a new value.
+        /// </summary>
+        /// <param name="initializer">The value initializer.</param>
+        internal LanguageVersionValue(in ValueInitializer initializer)
+            : base(
+                  initializer,
+                  initializer.Context.GetPrimitiveType(BasicValueType.Int32))
+        { }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary cref="Value.ValueKind"/>
+        public override ValueKind ValueKind => ValueKind.LanguageVersion;
+
+        #endregion
+
+        #region Methods
+
+        /// <summary cref="Value.Rebuild(IRBuilder, IRRebuilder)"/>
+        protected internal override Value Rebuild(
+            IRBuilder builder,
+            IRRebuilder rebuilder) =>
+            builder.CreateLanguageVersion(Location);
+
+        /// <summary cref="Value.Accept" />
+        public override void Accept<T>(T visitor) => visitor.Visit(this);
+
+        #endregion
+
+        #region Object
+
+        /// <summary cref="Node.ToPrefixString"/>
+        protected override string ToPrefixString() => "langVersion";
+
+        #endregion
+    }
+
+    /// <summary>
+    /// Represents an inline lanaguage statement.
+    /// </summary>
+    [ValueKind(ValueKind.LanguageEmit)]
+    public sealed class LanguageEmitValue : Value
```

Comment:
I guess we should make this class inherit from `MemoryValue` to avoid moving this value around due to possible side effects.

Reply:
Done.